mkosi/__init__.py

```diff
@@ -6948,15 +6954,19 @@ def run_qemu(args: CommandLineArguments) -> None:
         cmdline += ["-vga", "virtio"]
 
     if args.network_veth:
-        ensure_networkd(args)
-        # Use vt- prefix so we can take advantage of systemd-networkd's builtin network file for VMs.
-        ifname = f"vt-{virt_name(args)}"
-        # vt-<image-name> is the ifname on the host and is automatically picked up by systemd-networkd which
-        # starts a DHCP server on that interface. This gives IP connectivity to the VM. By default, QEMU
-        # itself tries to bring up the vt network interface which conflicts with systemd-networkd which is
-        # trying to do the same. By specifiying script=no and downscript=no, We tell QEMU to not touch vt
-        # after it is created.
-        cmdline += ["-nic", f"tap,script=no,downscript=no,ifname={ifname},model=virtio-net-pci"]
+        networkd_configured = ensure_networkd(args) == 0
+        if not networkd_configured:
+            # Fall back to usermode networking if the host doesn't have networkd (eg: Debian)
+            cmdline += ["-nic", f"user,model=virtio-net-pci"]
```

Comment:
I always find automatic fallback awkward, since a user might explicitly want networkd to be used or have no networking at all otherwise. should we make this an explicit option?

Reply:
There's a big red warning when networkd is enabled, that's should be enough? In most cases you don't want to change the default network management configuration of your distribution, as chances are it's not supported and things will break. Having things "just work" would be nice. The nice thing is that the fallback will _always_ work, regardless of system settings or anything else, and doesn't require any root  privileges, as it's entirely self-contained in qemu. So it's pretty safe to use.

Reply:
```
‣ Warning: 
            --network-veth requires systemd-networkd to be running to initialize the host interface of the
            veth link (`systemctl enable --now systemd-networkd`)")
```